scylla/tests/integration/query_result.rs

```diff
@@ -295,3 +307,131 @@ async fn execute_iter_serialization_error() {
         Err(PagerExecutionError::SerializationError(_))
     )
 }
+
+#[tokio::test]
+#[ntest::timeout(60000)]
+#[cfg(not(scylla_cloud_tests))]
+async fn should_expose_execution_info_on_exceptions() {
+    setup_tracing();
+    let res = test_with_3_node_cluster(
+        ShardAwareness::Unaware,
+        |proxy_uris, translation_map, mut running_proxy| async move {
+            let session: Session = SessionBuilder::new()
+                .known_node(proxy_uris[0].as_str())
+                .address_translator(Arc::new(translation_map))
+                .build()
+                .await
+                .unwrap();
+
+            let forge_error_rule = RequestRule(Condition::True, RequestReaction::forge().invalid());
+            running_proxy.running_nodes[0]
+                .change_request_rules(Some(vec![forge_error_rule.clone()]));
+            running_proxy.running_nodes[1]
+                .change_request_rules(Some(vec![forge_error_rule.clone()]));
+            running_proxy.running_nodes[2]
+                .change_request_rules(Some(vec![forge_error_rule.clone()]));
+
+            let mut query = Query::from("select * from foo");
+            let history_listener = Arc::new(HistoryCollector::new());
+            query.set_history_listener(history_listener.clone());
+            let err = session
+                .query_unpaged(query, &[])
+                .await
+                .expect_err("expecting error");
+            let structured_history: StructuredHistory = history_listener.clone_structured_history();
+
+            assert_matches!(
+                err,
+                ExecutionError::LastAttemptError(RequestAttemptError::DbError(_, _))
+            );
+            assert_eq!(structured_history.requests.len(), 1);
+            running_proxy
+        },
+    )
+    .await;
+
+    match res {
+        Ok(()) => (),
+        Err(err) => panic!("{}", err),
+    }
+}
```

Comment:
Isn't that mostly a `history` tests? What does it verify that tests in `integration/history.rs` don't?

Reply:
basically I wanted to test more details in errors, like Java driver does provide - but Rust driver don't expose them.
Somehow I have problems with finding similar tests - like also tracing tests I didn't notice already exist. Still I'm adapting to new IDE and ctrl+f mislead me.

Reply:
The structure of our tests (which test resides in which file) is unfortunately not the best :( We'd like to improve it - which is why I made my comments.
Feel free to ask if you can't find something that you suspect may already exist.
Nice way is to check for references to the function / method / field etc that you want to use in the test.
For `Session::get_tracing_info()` it does correctly find a reference in `integration/session.rs`:
![image](https://github.com/user-attachments/assets/41447e80-b615-4157-a9c3-64ff27245e0e)
specifically, in `test_get_tracing_info`. This function is in turn called by `test_tracing` in the same file, and there you can see our existing test cases for tracing.

Reply:
mostly my confusion is taken (again) out of python: test files usually start with `test_` so when I search for something and I see it in `test_xxx` file then I know it's in testing. Here when searching it is shown in `session.rs` which mislead me (I missed looking at the directories).